### PR TITLE
4.x Update associations.rst

### PR DESCRIPTION
**Pull request by joshv2**
Added an example of how to use query object with hasMany through associations.

### CODE REVIEW

**Review by Arhell — APPROVED** [review]

File: en/orm/associations.rst [hunk 1]
```diff
@@ -702,6 +702,17 @@ following models::
...
+and matching conditions for the main relationship table into your query object.
+The through table can then be referenced in other conditions such as a where 
+condition by designating the through table name before the field you are 
+filtering on.::
+
+    $query = $this->find('list', ['valueField' => 'studentFirstName', 'order' => 'students.id'])
+                        ->contain(['Courses'])
+                        ->matching('Courses')
+                        ->where(['CoursesMemberships.grade' => 'B']);
+
```
Comment (ndm2) [inline en/orm/associations.rst:715]: That's basically what https://book.cakephp.org/4/en/orm/retrieving-data-and-resultsets.html#filtering-by-associated-data-via-matching-and-joins describes.

Not sure if that should be repeated here, or if the existing docs should be expanded, but if it should be here, then it should IMHO have its own header, something regarding filtering and the join table, and a reference to the aforementioned general associated data filtering docs.

Comment (othercorey) [reply en/orm/associations.rst:715]: I'm not sure where a join table condition would go.

Comment (joshv2) [reply en/orm/associations.rst:715]: My thought is that all of the other types of associations in the associations documentation have an example of hjow to author a query using that type of association. This section introduces a new type/subtype of associations yet leaves the user wondering what the appropriate query object methods are to use to access the data in this type of association. The section referenced by @ndm2 does not explicitly mention "through" so the user is no more enlightened than before and even if this section did mention "through" there is no link to it where the "through" option is decribed on the associations page.

**Comment by othercorey** [comment]
@markstory Thought on wher/how to document this specific scenario?

**Comment by markstory** [comment]
I think the additions here are more directly related to the 'through' feature of BelongsToMany associations and the other page has similar examples, but not the same one.

I'm in favour of having both examples personally.

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**